src/validators/dataclass.rs

```diff
@@ -147,6 +152,10 @@ impl Validator for DataclassArgsValidator {
             ($args:ident, $get_method:ident, $get_macro:ident, $slice_macro:ident) => {{
                 // go through fields getting the value from args or kwargs and validating it
                 for (index, field) in self.fields.iter().enumerate() {
+                    let extra = Extra {
+                        field_name: Some(&field.name),
+                        ..extra
+                    };
```

Comment:
This seems necessary, but do we have any benchmarks that would pick up on performance impact of this? Not sure it matters.